docs(tutorials): tutorial private network with public gateway (#94)
scaleway/docs-content: tutorials/vpc-public-gateway-autoconfiguration/index.mdx

---
title: Deploying Instances in a Private Network and exposing them using a Public Gateway
description: This page shows how to deploy virtual Instances in a Private Network and expose them using a Public Gateway
excerpt: |

>   VPC consists of Private Networks and Public Gateways to create an isolated network environment for your instances, without exposing them on the public Internet. 

Reviewer: ```suggestion
  VPC consists of Private Networks and Public Gateways to create an isolated network environment for your i-Instances, without exposing them on the public Internet. 
```

totalTime: PT20M
  - network
brands:
  - vpc
keywords: cloud tutorials, cloud documentations, cloud instance, cloud hosting, network, virtual instance, private network, vpc, public gateway, virtual private cloud
---

## VPC - Overview

> VPC (Virtual Private Cloud) provides network functionalities for your Scaleway cloud. VPC products include [Private Networks](https://www.scaleway.com/en/private-networks/) , enabling you to build a virtual L2 network between your instances, and the [Public Gateway](https://www.scaleway.com/en/public-gateway/), enabling IP autoconfiguration of your Private Networks and their communication with the Internet.

Reviewer: ```suggestion
VPC (Virtual Private Cloud) provides network functionalities for your Scaleway cloud. VPC products include [Private Networks](https://www.scaleway.com/en/private-networks/), enabling you to build a virtual L2 network between your instances, and the [Public Gateway](https://www.scaleway.com/en/public-gateway/), enabling IP auto-configuration of your Private Networks and their communication with the Internet.
```

2. Check the status of your network interfaces using the `ip a` command. You see the details of the default Ethernet adapter of your Instance:

  ```
  2: ens2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether de:1c:a4:2b:d0:39 brd ff:ff:ff:ff:ff:ff
    inet 10.70.104.113/31 scope global dynamic ens2
       valid_lft 85983sec preferred_lft 85983sec
    inet6 2001:bc8:634:c38::1/64 scope global
       valid_lft forever preferred_lft forever
    inet6 fe80::dc1c:a4ff:fe2b:d039/64 scope link
       valid_lft forever preferred_lft forever
  ```

3. Go to the [VPC section](https://console.scaleway.com/vpc/private-networks) of the Scaleway console. The VPC creation page displays. 

  <Lightbox src="scaleway_vpc_splash.png" alt="" />

4. Click **Create a new Private Network**. The Private Network creation wizard displays. 

5. Enter the details of the new Private Network and click **Create a Private Network**:

    - Enter a **Name** for the Private Network
    - Select the region for the Private Network

    <Message type="note">

    Make sure to create the Private Network in the same region as your Instance.

    </Message>

> 6. The list of your Private Networks displays. Click on the Private Network's name to display its information. 

Reviewer: I think "The list of your private networks displays" should not be a step. Peut-être mettre la phrase avant la step  6 ?

> 7. Click the **Attached Resources** tab to see the list of instances attached to the Private Network. 

Reviewer: ```suggestion
7. Click the **Attached Resources** tab to see the list of Instances attached to the Private Network. 
```

8. Select your Instance from the drop down menu and click **Add Instance** to attach it to the Private Network:

  <Lightbox src="scaleway_add_instance.png" alt="" />

> 9. Use the `ip a` command on the Instance to verify the presence of the new ethernet interface: 

Reviewer: ```suggestion
9. Use the `ip a` command on the Instance to verify the presence of the new Ethernet interface: 
```

  ```
  3: ens5: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 02:00:00:00:5a:d8 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::ff:fe00:5ad8/64 scope link
       valid_lft forever preferred_lft forever
  ```

The Instance is now added to the Private Network but no IP address has been configured. You can configure it manually or use DHCP auto configuration with a Public Gateway. 

## Creating a Public Gateway

1. Click the [Public Gateways](https://console.scaleway.com/vpc/public-gateways) tab in the VPC section of the Scaleway console. The Public Gateways splash screen displays. Click **Create a Public Gateway**:

  <Lightbox src="scaleway_vpc_public_gateway_splash.png" alt="" />

> 2. The Public Gateway creation wizard displays. Enter the details of the new Public Gateway: 

Reviewer: Peut-être mettre "The Public Gateway creation wizard displays" avant la step 2

    - Choose the Availability Zone of the Public Gateway.
    - Select the Public Gateway offer
    - Select the Public Gateway IP address. To create a new IP, select **Allocate new IP** in the drop-down list. 
    - Enter a name and optional tags for the Public Gateway

3. Click **Create a Public Gateway**. The list of your Public Gateways displays.

  <Lightbox src="scaleway_vpc_public_gateway_list.png" alt="" />

4. Click **Manage** to display the Public Gateway Information. 

5. Click **+ Attach a new Private Network** to attach the previously created Private Network to the Public Gateway.

6. Select the Private Network from the drop-down list, then click **Attach Private Network**:

  <Lightbox src="scaleway_vpc_public_gateway_attach_network.png" alt="" /> 

> 7. Check if the IP address for the Private Network has been assigned to the instance using the DHCP server of the Private Gateway by running the `ip a` command: 

Reviewer: ```suggestion
7. Check if the IP address for the Private Network has been assigned to the Instance using the DHCP server of the Private Gateway by running the `ip a` command: 
```

  ```
  3: ens5: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 02:00:00:00:5a:d8 brd ff:ff:ff:ff:ff:ff
    inet 192.168.0.2/24 brd 192.168.0.255 scope global dynamic ens5
       valid_lft 3513sec preferred_lft 3513sec
    inet6 fe80::ff:fe00:5bea/64 scope link 
       valid_lft forever preferred_lft forever
  ```

  You can see the private IP address `192.168.0.2` has been configured automatically on the interface.

  <Message type="note">

>   Once the Instance is added to the Private Network and auto configured through DHCP, it cannot be accessed through its public IPv4 address anymore. Later in this tutorial, we explain how to configure port forwarding to access your instances through the Public Gateway's public IPv4 address.

Reviewer: ```suggestion
  Once the Instance is added to the Private Network and auto configured through DHCP, it cannot be accessed through its public IPv4 address anymore. Later in this tutorial, we explain how to configure port forwarding to access your Instances through the Public Gateway's public IPv4 address.
```
